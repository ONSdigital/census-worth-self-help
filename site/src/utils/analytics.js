export default class Analytics {
  static trackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    } else {
      console.debug(`Track event not set : ${category} - ${action} - ${name} - ${value}`)
    }
  }

  static trackSiteSearch(keyword, category, searchCount) {
    if (window._paq) {
      window._paq.push(["trackSiteSearch", keyword, category, searchCount])
    } else {
      console.debug(`Track site search : ${keyword} - ${category} - ${searchCount}`)
    }
  }

  static setPageType(pageType) {
    if (typeof window !== `undefined` && window._paq) {
      window._paq.push(["setCustomDimension", 1, pageType])
    }
  }
}