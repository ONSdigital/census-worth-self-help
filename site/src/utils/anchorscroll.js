export default (history, timeout = 1000) => {
  let observer
  let timeoutId

  const reset = () => {
    if (timeoutId) {
      clearTimeout(timeoutId)
      timeoutId = null
    }

    if (observer) {
      observer.disconnect()
    }
  }

  const convertTextToLink = text => {
    return text.toLowerCase().replace(/\s/g, "-")
  }

  const createScrollToElement = hash => {
    return () => {
      let headers = []
      let tags = ["h1", "h2", "h3"]

      tags.forEach(tag => {
        headers = headers.concat(Array.from(document.getElementsByTagName(tag)))
      })

      let element = headers.find(
        header => convertTextToLink(header.textContent) === hash
      )

      if (element) {
        element.scrollIntoView()
        reset()
        return true
      }

      return false
    }
  }

  function scroll(hash) {
    if (typeof hash !== "string") {
      return
    }

    const anchorText = hash.slice(1).toLowerCase()

    if (!anchorText) {
      return
    }

    const scrollToElement = createScrollToElement(anchorText)

    setTimeout(() => {
      if (scrollToElement()) {
        return
      }

      if (window.MutationObserver) {
        observer = new MutationObserver(scrollToElement)

        observer.observe(document, {
          attributes: true,
          childList: true,
          subtree: true
        })
      }

      timeoutId = setTimeout(reset, timeout)
    })
  }

  // listen for changes
  history.listen(({ location, action }) => {
    if (timeoutId) {
      reset()
    }

    if (action !== "POP") {
      return
    }

    if (location.hash) {
      scroll(location.hash)
    }
  })

  // also trigger on load
  if (history.location.hash) {
    scroll(history.location.hash)
  }
}
