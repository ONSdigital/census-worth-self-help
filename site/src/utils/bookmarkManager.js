export default class BookmarkManager {
  constructor() {
    this.bookmarks = JSON.parse(localStorage.getItem("bookmarks"))
    if (this.bookmarks === null) {
      this.bookmarks = []
    }
  }

  getTopBookmarks() {
    return this.bookmarks
  }

  isPageBookmarked(page) {
    return this.bookmarks.includes(page)
  }

  bookmarkPage(page) {
    if (!this.isPageBookmarked(page)) {
      this.bookmarks.push(page)
    }
    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }

  unBookmarkPage(page) {
    this.bookmarks = this.bookmarks.filter(function(bookmark) {
      return bookmark !== page
    })

    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }
}
