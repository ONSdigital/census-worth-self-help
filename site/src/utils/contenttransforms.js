var sanitizeHtml = require("sanitize-html")

export const transformSources = htmlString => {
  let videoPath = process.env.GATSBY_ASSETS_PATH
    ? process.env.GATSBY_ASSETS_PATH
    : ""
  if (!htmlString) {
    return ""
  }
  return htmlString.replace(/{{TARGET_ASSETS_SRC}}/g, videoPath)
}

export const htmlSanitize = htmlString => {
  return sanitizeHtml(htmlString, {
    allowedTags: sanitizeHtml.defaults.allowedTags.filter(function(tag) {
      return tag !== "iframe"
    }).concat([
      "video",
      "audio",
      "img",
      "source",
      "h1",
      "h2"
    ]),
    allowedAttributes: {
      a: ["href", "name", "target"],
      img: ["src", "title"],
      source: ["src", "type"],
      video: ["controls"],
      audio: ["controls"]
    }
  })
}
