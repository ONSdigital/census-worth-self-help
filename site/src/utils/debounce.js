import _ from "lodash"

// debounce with default delay, or return sync function if delay exactly zero (for unit tests)
export default (fn, delay) => {
  return (delay === 0)
    ? fn
    : _.debounce((...args) => fn(...args), delay || 1000)
}