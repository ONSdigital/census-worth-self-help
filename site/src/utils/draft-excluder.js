const READY_FOR_PRODUCTION = "Ready for Live Site"

// Articles in draft should not be deployed to production. An article is draft if the draftreason for the article
// is anything other than "Ready for production" - see ONS-359.
//
// We also use this Draft Excluder concept to not show articles that are for the call centre only.
//
// Initially this was done using the draft tag. This technique might be removed in the future.
exports.excludeDraftArticle = node => {
  return Boolean(
    process.env.EXCLUDE_DRAFTS && (
      (
        node.frontmatter.tags &&
        node.frontmatter.tags.includes("draft")
      ) || (
        node.frontmatter.draftreason &&
        node.frontmatter.draftreason !== READY_FOR_PRODUCTION
      ) || node.frontmatter.cconly
    )
  )
}
