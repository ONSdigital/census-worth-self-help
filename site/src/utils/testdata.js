export const data = {
  edges: [
    {
      node: {
        frontmatter: {
          title: "test Article 1",
          description: "testDescription",
          priority: 1,
          date: new Date()
        },
        fields: {
          pagename: "test-article-1",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 2",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular", "webchat"]
        },
        fields: {
          pagename: "test-article-2",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 3",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-3",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 4",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["draft"]
        },
        fields: {
          pagename: "test-article-4",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 5",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-5",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 6",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-6",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Directory 7",
          description: "testDescription",
          priority: 0,
          date: new Date()
        },
        fields: {
          pagename: "test-directory-1",
          collection: "directories"
        }
      }
    }
  ]
}

export const articleNode = data.edges[0].node
export const webchatArticleNode = data.edges[1].node
export const directoryNode = data.edges[2].node
export const draftNode = data.edges[3].node
export const popularList = {
  edges: [data.edges[1], data.edges[2], data.edges[4], data.edges[5]]
}
export const articleList = { edges: data.edges.slice(0, 6) }
