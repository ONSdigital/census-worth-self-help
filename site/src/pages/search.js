import React from "react"
import { graphql } from "gatsby"
import { css } from "@emotion/core"
import Layout from "../components/layout"
import { Index } from "elasticlunr"
import TabList from "../components/tablist"
import PageTitle from "../components/pagetitle"
import PaginationBar from "../components/paginationbar"

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faSearch } from "@fortawesome/free-solid-svg-icons"
import { PaginationObject } from "../utils/pagination"

const escapeStringRegexp = require("escape-string-regexp")
const minimumSearchString = 3

export default class Search extends React.Component {
  constructor(props) {
    super(props)

    let paginationObject = new PaginationObject()
    this.state = {
      query: ``,
      results: [],
      paginationObject: paginationObject
    }
    this.data = props.data
    this.updateSearchResults = this.updateSearchResults.bind(this)
    this.updatePagination = this.updatePagination.bind(this)
  }

  updatePagination(pageTarget) {
    this.state.paginationObject.goToPage(pageTarget)
    // update state to get page to rerender
    this.setState({
      paginationObject: this.state.paginationObject
    })
  }

  updateSearchResults(evt) {
    this.state.paginationObject.goToPage(0)

    const query = evt.target.value
    this.index = this.index
      ? this.index
      : Index.load(this.data.siteSearchIndex.index)

    // we use a weighted priority of the search strings, this should be calibrated based on user feedback
    this.index.search(query, {
      fields: {
        title: { boost: 4 },
        author: { boost: 4 },
        tags: { boost: 4 },
        description: { boost: 2 },
        body: { boost: 1 }
      }
    })
    this.setState({
      query,
      results: this.index
        .search(query, {})
        .map(({ ref }) => this.index.documentStore.getDoc(ref))
    })
  }

  /*
   Takes a string and a regex of the form /(bold1|bold2|bold3)/ig
   Creates a jsx renderable list of strings and bold elements  
  */
  static replacePatternToBold(text, pattern) {
    const splitText = text.split(pattern)
    const matches = text.match(pattern)
    if (splitText.length <= 1) {
      return text
    }

    return splitText.reduce((arr, element) => {
      if (!element) return arr
      if (matches.includes(element)) {
        return [...arr, <strong>{element}</strong>]
      }
      return [...arr, element]
    }, [])
  }

  static getTagsAsString(tags) {
    if (!tags) {
      return ""
    }
    if (Array.isArray(tags)) {
      tags = tags.join(", ")
    }
    return tags
  }

  static stripHTML(html) {
    var doc = new DOMParser().parseFromString(html, "text/html")
    return doc.body.textContent || ""
  }

  static highlightNode(node, query) {
    // normalizes the string into a list of words (space seperated)
    let splitQuery = query
      .trim()
      .toLowerCase()
      .split(" ")
      .filter(str => str)

    // fetches all properties we wish to highlight in order of precedence
    let properties = [
      node.frontmatter.author,
      node.frontmatter.description,
      Search.getTagsAsString(node.frontmatter.tags),
      Search.stripHTML(node.html)
    ]

    // finds first property which includes a query word
    let highlightableText = properties.find(property => {
      if (!property) {
        return false
      }
      return splitQuery.find(queryWord =>
        property.toLowerCase().includes(queryWord)
      )
    })

    // if a property is found we bould the query words
    if (highlightableText !== undefined) {
      let pattern = new RegExp(
        "(" + splitQuery.map(x => escapeStringRegexp(x)).join("|") + ")",
        "ig"
      )
      highlightableText = Search.replacePatternToBold(
        highlightableText,
        pattern
      )
    } else {
      // otherwise default to description
      highlightableText = node.frontmatter.description
    }

    // we store the result in a new attribute, which articletab checks for.
    node.highlightedText = highlightableText
  }

  render() {
    // the search object is given to the top bar to control search
    let searchObject = {
      updateFunction: this.updateSearchResults,
      query: this.state.query
    }

    // fetch the data edges corresponding to the search results
    let edges = this.state.paginationObject
      .filterResults(this.state.results)
      .map(result => {
        let edge = this.data.allMarkdownRemark.edges.find(
          edge => edge.node.frontmatter.title === result.title
        )
        Search.highlightNode(edge.node, this.state.query)
        return edge
      })

    // A user doesn't count as searching unless they've typed a minimum amount
    let searching =
      this.state.query.length >= minimumSearchString || edges.length > 0

    return (
      <Layout title="Search" searchObject={searchObject}>
        <PageTitle>
          <FontAwesomeIcon
            icon={faSearch}
            css={css`
              padding: 5px;
            `}
          />
          {searching && edges.length > 0 && (
            <div>
              {this.state.results.length} results for "{this.state.query}"
            </div>
          )}
          {searching && edges.length === 0 && (
            <div>Sorry no results for "{this.state.query}"</div>
          )}
          {!searching && <div>Begin typing to search</div>}
        </PageTitle>
        <TabList elements={edges} />
        {this.state.results.length !== 0 && (
          <PaginationBar
            total={this.state.results.length}
            paginationObject={this.state.paginationObject}
            clickFunction={this.updatePagination}
            onPageCount={edges.length}
          />
        )}
      </Layout>
    )
  }
}

export const query = graphql`
  query {
    siteSearchIndex {
      index
    }

    allMarkdownRemark(filter: { fields: { collection: { eq: "articles" } } }) {
      edges {
        node {
          html
          ...BaseArticleFields
        }
      }
    }
  }
`
