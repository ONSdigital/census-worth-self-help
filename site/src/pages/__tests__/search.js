import React from "react"
import renderer from "react-test-renderer"
import Search from "../search"
import { getSuggestedEdges } from "../search"
import { render } from "react-testing-library"
import { articleList, articleNode, popularList } from "../../utils/testdata"
import ReactDOMServer from 'react-dom/server';

describe("Search", () => {
  it("renders correctly", () => {
    const data = { allMarkdownRemark : articleList }
    const tree = renderer.create(<Search data={data}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("bold-pattern function works correctly", () => {
    expect(ReactDOMServer.renderToStaticMarkup(
    	Search.replacePatternToBold("we are bolding test", new RegExp("(test)", 'ig'))
    	)).toEqual("we are bolding <strong>test</strong>")

    expect(ReactDOMServer.renderToStaticMarkup(
    	Search.replacePatternToBold("we are bolding test and best", new RegExp("(test|best)", 'ig'))
    	)).toEqual("we are bolding <strong>test</strong> and <strong>best</strong>")

    expect(ReactDOMServer.renderToStaticMarkup(
    	Search.replacePatternToBold("we are bolding nothing", new RegExp("(something)", 'ig'))
    	)).toEqual("we are bolding nothing")

    expect(ReactDOMServer.renderToStaticMarkup(
    	Search.replacePatternToBold("we are bolding just this", new RegExp("(just this|notthis)", 'ig'))
    	)).toEqual("we are bolding <strong>just this</strong>")
  })

  it("tag stripping", () => {
    expect(Search.stripHTML("<p><br />hi<br /> <h1>hello</h1>")).toEqual("hi hello")
  })

  it("highlight function on node works correctly", () => {
    let node = articleNode
    node.frontmatter.tags = ['a', 'b', 'hello', 'd']
    node.frontmatter.description = "this is my beautiful description"
    node.frontmatter.author = "Napolean"

    // search description
    Search.highlightNode(node, 'beautiful')
    expect(ReactDOMServer.renderToStaticMarkup(node.highlightedText)).toEqual("this is my <strong>beautiful</strong> description")

    // search tags
    Search.highlightNode(node, 'hello')
    expect(ReactDOMServer.renderToStaticMarkup(node.highlightedText)).toEqual("a, b, <strong>hello</strong>, d")

    // search author
    Search.highlightNode(node, 'Napolean')
    expect(ReactDOMServer.renderToStaticMarkup(node.highlightedText)).toEqual("<strong>Napolean</strong>")

    // search capitalisied tags
    Search.highlightNode(node, 'HELLO')
    expect(ReactDOMServer.renderToStaticMarkup(node.highlightedText)).toEqual("a, b, <strong>hello</strong>, d")

    // search missing, default to description
    Search.highlightNode(node, 'Waldo')
    expect(ReactDOMServer.renderToStaticMarkup(node.highlightedText)).toEqual("this is my beautiful description")

    // search two words in description
    Search.highlightNode(node, 'description beautiful')
    expect(ReactDOMServer.renderToStaticMarkup(node.highlightedText)).toEqual("this is my <strong>beautiful</strong> <strong>description</strong>")
  })

  it("get suggested nodes should find popular nodes", () => {
    expect(getSuggestedEdges(articleList.edges)).toEqual( popularList.edges.slice(0, 3) )
  })

})