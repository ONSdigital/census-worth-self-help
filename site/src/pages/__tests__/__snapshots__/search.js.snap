// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "vkixka",
      "next": undefined,
      "styles": "
        height: 100vh;
        display: flex;
        flex-direction: column;
      ",
    }
  }
>
  <header
    css={
      Object {
        "map": undefined,
        "name": "181ra4m",
        "next": undefined,
        "styles": "
        width: 100vw;
        
    background: linear-gradient(to left, rgb(0,61,89), #33637a);
  ;
      ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1thgah9",
          "next": undefined,
          "styles": "
            display: flex;
          ",
        }
      }
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "1uk2jmg",
            "next": undefined,
            "styles": "
              padding: 10px 15px 14px 15px;
              flex-grow: 1;
              display: flex;
            ",
          }
        }
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "10cznun",
              "next": undefined,
              "styles": "
                
    display: flex;
    justify-content: center;
    flex-direction: column;
  
                position: relative;
                left: 5px;
              ",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            css={
              Object {
                "map": undefined,
                "name": "5duw79",
                "next": undefined,
                "styles": "
                  position: absolute;
                  color: #6e6e6e;
                ",
              }
            }
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <input
          autoFocus={true}
          className="Notification-heading-Style"
          css={
            Object {
              "map": undefined,
              "name": "1f3qd7n",
              "next": undefined,
              "styles": "
                border: 0;
                border-radius: 16px;
                background-color: rgb(250, 250, 250);
                flex-grow: 1;
                padding-left: 25px;
              ",
            }
          }
          data-testid="search-box"
          maxLength="60"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        Menu
      </div>
    </div>
  </header>
  <div
    css={
      Object {
        "map": undefined,
        "name": "o9ztju",
        "next": undefined,
        "styles": "
          
    margin: 0 auto;
    max-width: 700px;
  
          overflow: scroll;
          width: 100vw;
          display: flex;
          flex-direction: column;
          flex-grow: 1;
        ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1k0aln6",
          "next": undefined,
          "styles": "
            flex-grow: 1;
          ",
        }
      }
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "12ew49b",
            "next": undefined,
            "styles": "
        
    padding: 5px 15px;
  
      ",
          }
        }
      >
        <h1
          className="Article-Title-Style"
          css={
            Object {
              "map": undefined,
              "name": "1xcgh0k",
              "next": undefined,
              "styles": "
          margin-bottom: 0px;
          display: flex;
        ",
            }
          }
          data-testid="search-result-title"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            css={
              Object {
                "map": undefined,
                "name": "1kk0sdj",
                "next": undefined,
                "styles": "
              padding: 5px;
            ",
              }
            }
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <div>
            Begin typing to search
          </div>
        </h1>
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "1p7jpnp",
            "next": undefined,
            "styles": "
        
    padding: 15px 0px;
  ;
      ",
          }
        }
      />
    </div>
    <div
      css={
        Object {
          "map": undefined,
          "name": "ej75ms",
          "next": undefined,
          "styles": "
        
    padding: 30px 0px;
  
        width: 100%;
        padding: 10px 0px;
        background-color: rgb(236, 236, 237);
        display: flex;
        min-height: 100px;
        flex-direction: column;
        justify-content: center;
        flex-shrink: 0;
      ",
        }
      }
    >
      <img
        alt="Footer logo"
        css={
          Object {
            "map": undefined,
            "name": "1v6eegl",
            "next": undefined,
            "styles": "
          margin-bottom: 50px;
          margin-top: 50px;
        ",
          }
        }
        src="/footer_logo.svg"
      />
      <div
        className="Footer-Style"
      >
        Â© Office for National Statistics 2019
      </div>
    </div>
  </div>
</div>
`;
